pkg/controller/core/clusterqueue_controller_test.go

```diff
@@ -65,6 +65,7 @@ func TestUpdateCqStatusIfChanged(t *testing.T) {
 					Reason:  "FlavorNotFound",
 					Message: "Can't admit new workloads; some flavors are not found",
 				}},
+				PendingWorkloadsStatus: &kueue.ClusterQueuePendingWorkloadsStatus{},
```

Comment:
The feature is disabled here so I expect `nil` to be returned actually. Let's adjust the implementation and add a unit test. I suggest to expand this test, but adding a dedicated test case is also fine. If we expand this one the following will need to change:
- we call `Start` on the Reconciller
- add extra params to the test case struct: maxCount, updateInterval (existing test cases don't need updates)
- add test cases for MaxCount=2 & updateInterval=10ms. 
- When MaxCount>0 the test case should loop await until the snapshot is available by checking `qManager.GetSnapshot(cqName)` periodically, before calling `updateCqStatusIfChanged`

Reply:
added separate test cases https://github.com/kubernetes-sigs/kueue/pull/1069/commits/3b30bd48c7e9b9de73d3d87d00bd3830281a8819